compression/tools/test_slow_drain.py

```
@@ -0,0 +1,425 @@
+    m = re.search(r"(?im)^content-encoding:\s*(\S+)", headers)
+    got = m.group(1) if m else None
+    if got != coding:
+        raise RuntimeError(f"{path}: Content-Encoding {got!r}, "
+                           f"wanted {coding!r} (status {status!r}, "
+                           f"{len(body)}B body)")
+    return body
+
+
+def decode(coding: str, blob: bytes) -> bytes:
+    """Decompress with the reference CLI; nonzero exit = truncation."""
+    r = subprocess.run(CODINGS[coding]["decode"], input=blob,
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)